### PR TITLE
149 how to add new modules

### LINKED ISSUE #149: Create specification for how to add new modules
# Summary
A written specification should be created on how to add new modules to the application, which parts a module need to implement etc.

# Examples
n/a

# Context
Should make it easy for someone to add additional modules without looking to much at the current source code and how everything works together.

### PR DESCRIPTION
**Pull request by Blinningjr**
# Proposed changes
* Added a automation module template
* Added development documentation on how to implement a automation module.

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

- Read the documentation and the template to see if it makes sense.

# Related issue
Fixes #149 

# Checkboxes
- [x] I have run the unit tests with `pytest`
- [x] I formatted the changes with `./scripts/formatter.sh`

### CODE REVIEW

File: docs/automation_module.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+The files that needs to be updated are 
+
+```
+requirements.txt
+config/language_releases.yaml
+config/nlp_models.yaml
+```
+
+In `requirments.txt` a link to the releases NLP models must be added or updated.
```
Comment (Aleman778) [inline docs/automation_module.md:19]: Should be requirements.txt

File: docs/automation_module.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+```
+
+### prepare
+
+This methods job is do all the preparations needed to run the `execute` method, but this method should only be called once for each execution of the module. Thus all the preparations that has to be done again when a followup question has been answered should be in the `prepare_processed` method. 
+
+### prepare\_processed
+
+A requirement of this method is that the NLP model has parsed the input into the following 4 parameter `to, when, body, sender` before calling this method.
+* `to` is the list of contacts that user want to send a email to for example.
```
Comment (Aleman778) [inline docs/automation_module.md:74]: an email

File: docs/automation_module.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+* `when` is the time, in the schedule it is a dict with the start and end time of an event.
+* `body` can be anything.
+* `sender` is information of the user, like email information in the email example.
+
+The job of this method is to take the parsed input form the NLP model and process it so that the `execute` method can run. This can be thinks like finding en email related to a name or giving followup questions to the user about missing information.
```
Comment (Aleman778) [inline docs/automation_module.md:79]: Second sentence. Correction: This can be _things_ like ...

**Review by Aleman778 — CHANGES_REQUESTED** [review]
Looks good! Some small errors I found.

**Review by Aleman778 — APPROVED** [review]

**Review by viktorfrom — APPROVED** [review]

**Pull request merged**